pkg/storage/replica.go

```diff
@@ -1519,6 +1519,18 @@ func (ec *endCmds) doneLocked(
 					// create a transaction record with WriteTooOld set.
 					key := keys.TransactionKey(header.Key, *cr.txnID)
 					cr.txn = roachpb.Span{Key: key}
+				case *roachpb.ScanRequest:
+					resp := br.Responses[i].GetInner().(*roachpb.ScanResponse)
+					if ec.ba.Header.MaxSpanRequestKeys != 0 &&
+						ec.ba.Header.MaxSpanRequestKeys == int64(len(resp.Rows)) {
+						// If the scan requested a limited number of results and we hit the
```

Comment:
My initial reaction here was "why do we not restrict the range always to the last result found, regardless of whether the limit was reached"?

I *think* this is because if the query requests 10 rows but we only find 5, we still want to mark the entire range as touched for ... consistency reasons or something ... (perhaps to ensure that matching rows that are created after the query is run are still covered by this ts cache entry?)

So there, I think that whatever the reason is why we don't restrict the endkey in every case needs to be put in the comment because it's not trivial.